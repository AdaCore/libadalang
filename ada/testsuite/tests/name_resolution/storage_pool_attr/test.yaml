driver: name-resolution
input_sources: [storage_pool_attr.adb]
with_default_project: False
