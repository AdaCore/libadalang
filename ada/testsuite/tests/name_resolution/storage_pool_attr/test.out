Analyzing storage_pool_attr.adb
###############################

Resolving xrefs for node <ObjectDecl ["S"] storage_pool_attr.adb:10:4-10:71>
****************************************************************************

Expr: <AttributeRef storage_pool_attr.adb:10:8-10:52>
  references: <DefiningName s-stopoo.ads:42:9-42:26>
  type:       None
Expr: <DottedName storage_pool_attr.adb:10:8-10:46>
  references: <DefiningName s-stopoo.ads:42:9-42:26>
  type:       None
Expr: <DottedName storage_pool_attr.adb:10:8-10:28>
  references: <DefiningName s-stopoo.ads:39:9-39:29>
  type:       None
Expr: <Id "System" storage_pool_attr.adb:10:8-10:14>
  references: <DefiningName system.ads:37:9-37:15>
  type:       None
Expr: <Id "Storage_Pools" storage_pool_attr.adb:10:15-10:28>
  references: <DefiningName s-stopoo.ads:39:9-39:29>
  type:       None
Expr: <Id "Root_Storage_Pool" storage_pool_attr.adb:10:29-10:46>
  references: <DefiningName s-stopoo.ads:42:9-42:26>
  type:       None
Expr: <Id "Class" storage_pool_attr.adb:10:47-10:52>
  references: None
  type:       None
Expr: <AttributeRef storage_pool_attr.adb:10:56-10:70>
  references: None
  type:       <ClasswideTypeDecl ["Root_Storage_Pool"] s-stopoo.ads:42:4-43:59>
Expr: <Id "B" storage_pool_attr.adb:10:56-10:57>
  references: <DefiningName storage_pool_attr.adb:7:4-7:5>
  type:       <TypeDecl ["A"] storage_pool_attr.adb:5:4-5:33>
Expr: <Id "Storage_Pool" storage_pool_attr.adb:10:58-10:70>
  references: None
  type:       None

Resolving xrefs for node <CallStmt storage_pool_attr.adb:13:4-13:29>
********************************************************************

Expr: <CallExpr storage_pool_attr.adb:13:4-13:28>
  references: <DefiningName s-stopoo.ads:46:14-46:22>
  type:       None
Expr: <DottedName storage_pool_attr.adb:13:4-13:14>
  references: <DefiningName s-stopoo.ads:46:14-46:22>
  type:       None
Expr: <Id "S" storage_pool_attr.adb:13:4-13:5>
  references: <DefiningName storage_pool_attr.adb:10:4-10:5>
  type:       <ClasswideTypeDecl ["Root_Storage_Pool"] s-stopoo.ads:42:4-43:59>
Expr: <Id "Allocate" storage_pool_attr.adb:13:6-13:14>
  references: <DefiningName s-stopoo.ads:46:14-46:22>
  type:       None
Expr: <Id "Add" storage_pool_attr.adb:13:16-13:19>
  references: <DefiningName storage_pool_attr.adb:8:4-8:7>
  type:       <TypeDecl ["Address"] system.ads:67:4-67:28>
Expr: <Int storage_pool_attr.adb:13:21-13:23>
  references: None
  type:       <TypeDecl ["Universal_Int_Type_"] __standard:119:3-119:45>
Expr: <Int storage_pool_attr.adb:13:25-13:27>
  references: None
  type:       <TypeDecl ["Universal_Int_Type_"] __standard:119:3-119:45>


Done.
